### ISSUE TITLE
[2.7] Computed property is not being updated when built with Composition API and lib mode

### ISSUE THREAD

**Issue by andresilva-cc** [vuejs/vue#12904]
### Version
2.7.14

### Reproduction link
[github.com](https://github.com/andresilva-cc/vue-composition-api-reactive-bug)

### Steps to reproduce
**component directory**

1. Install the dependencies with `yarn`

2. Build the component with `yarn build`

**client directory**

1. Install the dependencies with `yarn`

2. Run the project with `yarn dev`

3. Open the URL in the browser

4. Try to change the input value and see that the computed length does not change (both on the template and on Vue Devtools)

### What is expected?
It is expected that when a computed dependency changes, the computed value changes and updates the template

### What is actually happening?
The computed property is not being changed after the first time nor the template is being updated

---
If the component is built with Options API it works normally.

**Comment by posva** [comment]
You have to `dedupe` vue in your client to avoid two copies of vue being included.

This happens because the component is loading vue from the component folder while the client is loading its own vue version. You have to ensure they both load the same vue copy. In practice, this doesn't happen when you install the input-test library as an npm package, here it happens because it's a local link.

You can find more information in issues like

- https://github.com/vuejs/core/issues/2600
- https://github.com/vuejs/core/issues/6984

**Comment by andresilva-cc** [comment]
@posva Actually this DOES happen when we install the package as an npm package. Our situation is that we are building a component library which is published to GitHub Packages npm registry and when we install it to another project this issue occurs.

I made the reproduction with a local link just so that the problem would be visible, as the packages we publish are private to the company only.

Note that on the `vite.config.js` of the component we are deduping Vue and also externalizing it, therefore it is not being included in the final build, so I don't think that's the problem.

I also said that when we build the component with Options API it works normally, so if two copies of Vue were to be the problem, then it would not matter if we are using Composition or Options API, the problem would happen anyway.

**Comment by andresilva-cc** [comment]
@posva Please if you could reopen this :)

**Comment by posva** [comment]
I did take the time to test the different solutions I mentioned and they all worked so you might have a misconfiguration somewhere else

**Comment by andresilva-cc** [comment]
Well, the reproduction is really straightforward, we kept it at the minimum possible, so after all your tests do you see something off with the reproduction?

I just can't see why it is failing, and the fact that if we refactor the component to Options API just makes me think that maybe this is a bug in Vue.

Maybe I can publish a test package to npm just to make the reproduction more "reliable", and include both versions of the component, with Composition and Options API.

**Comment by andresilva-cc** [comment]
@posva This issue has been resolved :)

It indeed appears to be a bug in Vue, as projects running the latest version (`2.7.14`) were working just fine, while the two projects that I was testing on, which were running `2.7.10` and `2.7.13`, were not working. I've just updated both `vue` and `vue-template-compiler` to `2.7.14` and it started to work.

Again, I didn't need to change anything in the component library project, the issue was with the client apps.
